module/zfs/zil.c

```diff
@@ -1253,6 +1267,16 @@ zil_lwb_write_done(zio_t *zio)
 	 * written out.
 	 */
 	if (zio->io_error != 0) {
+		zil_commit_waiter_t *zcw;
+		for (zcw = list_head(&lwb->lwb_waiters); zcw != NULL;
+		    zcw = list_next(&lwb->lwb_waiters, zcw)) {
+			mutex_enter(&zcw->zcw_lock);
+			ASSERT(list_link_active(&zcw->zcw_node));
+			ASSERT3P(zcw->zcw_lwb, ==, lwb);
+			zcw->zcw_zio_error = zio->io_error;
```

Comment:
AFAIK this is the first place where we might set `zcw_zio_error` => if the `zcw` is zero-initialized, we could `ASSERT3S(zcw->zcw_zio_error, ==, 0);`

Reply:
> if the zcw is zero-initialized

it is in `zil_alloc_commit_waiter`.. further, we already rely on this in the flush function (with these changes):
```
VERIFY(zcw->zcw_zio_error == 0);
```
so I think we could do this verification here too.